fluss-common/src/main/java/com/alibaba/fluss/row/encode/iceberg/IcebergBinaryRowWriter.java

```diff
@@ -0,0 +1,390 @@
+            return;
+        }
+
+        // Iceberg stores timestamps as microseconds since epoch
+        ensureCapacity(8);
+        long micros = value.getMillisecond() * 1000L + (value.getNanoOfMillisecond() / 1000L);
+        segment.putLong(cursor, micros);
+        cursor += 8;
+    }
+
+    public void writeTimestampLtz(int pos, TimestampLtz value, int precision) {
+        if (value == null) {
+            setNullAt(pos);
+            return;
+        }
+
+        // Iceberg stores timestamptz as microseconds since epoch in UTC
+        ensureCapacity(8);
+        long epochMillis = value.getEpochMillisecond();
+        int nanoOfMilli = value.getNanoOfMillisecond();
+        long totalMicros = epochMillis * 1000L + (nanoOfMilli / 1000L);
+        segment.putLong(cursor, totalMicros);
+        cursor += 8;
+    }
+
+    private void ensureCapacity(int neededSize) {
+        if (segment.size() < cursor + neededSize) {
+            grow(cursor + neededSize);
+        }
+    }
+
+    private void grow(int minCapacity) {
+        int oldCapacity = segment.size();
+        int newCapacity = Math.max(oldCapacity * 2, minCapacity);
+        setBuffer(Arrays.copyOf(buffer, newCapacity));
+    }
+
+    private void setBuffer(byte[] buffer) {
+        this.buffer = buffer;
+        this.segment = MemorySegment.wrap(buffer);
+    }
+
+    private void extractFromFlussSegments(BinaryString input, byte[] target) {
+        // Convert Fluss MemorySegments to byte array
+        MemorySegment[] segments = input.getSegments();
+        int offset = input.getOffset();
+        int length = input.getSizeInBytes();
+
+        if (segments.length == 1) {
+            segments[0].get(offset, target, 0, length);
+        } else {
+            copyFromFlussSegments(segments, offset, target, 0, length);
+        }
+    }
+
+    private void copyFromFlussSegments(
+            MemorySegment[] segments, int offset, byte[] target, int targetOffset, int length) {
+        int remaining = length;
+        int currentOffset = offset;
+        int currentTargetOffset = targetOffset;
+
+        for (MemorySegment segment : segments) {
+            if (remaining <= 0) {
+                break;
+            }
+
+            int segmentSize = segment.size();
+            if (currentOffset >= segmentSize) {
+                currentOffset -= segmentSize;
+                continue;
+            }
+
+            int availableInSegment = segmentSize - currentOffset;
+            int toCopy = Math.min(remaining, availableInSegment);
+
+            // Copy from MemorySegment to byte array
+            segment.get(currentOffset, target, currentTargetOffset, toCopy);
+
+            remaining -= toCopy;
+            currentTargetOffset += toCopy;
+            currentOffset = 0;
+        }
+    }
+
+    private static int calculateInitialSize(int arity) {
+        // Conservative estimate for Iceberg format
+        return 64 + (arity * 16); // Allow for variable length fields
```

Comment:
Why use `64 + (arity * 16);`? I'd suggest to use `8 + (arity * 8);` for space save. No need to allocate a larger byte array.